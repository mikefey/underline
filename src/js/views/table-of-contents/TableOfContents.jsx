import React from 'react';
import PropTypes from 'prop-types';
import Header from 'ui/header/Header.jsx';
import CloseButton from 'ui/close-button/CloseButton.jsx';
import TableOfContentsButton from './TableOfContentsButton.jsx';
import style from './TableOfContents.scss';


/**
 * Table of contents view component
 * @param {Object} props - The component props
 */
const TableOfContents = (props) => {
  const data = props.tableOfContentsState.data;
  const wrapperClass = props.tableOfContentsState.show
    ? style.wrapperShowing
    : style.wrapper;


  /**
   * Renders button components
   * @returns {Array<ReactElement>} Buttons for each chapter
   */
  const renderButtons = () => {
    let buttons = [];

    if (data) {
      buttons = data.map((item) => {
        const clickHandler = () => {
          props.tableOfContentsButtonClicked(item.href);
          props.hideTableOfContents();
        };

        return (
          <TableOfContentsButton
            data={item}
            key={`toc-section-button-${item.href}`}
            clickHandler={clickHandler}
          />
        );
      });
    }

    return buttons;
  };

  return (
    <div className={wrapperClass}>
      <Header
        rightButton={
          <CloseButton
            clickHandler={props.hideTableOfContents}
          />
        }
        text='Table Of Contents'
      />
      <div className={style.buttonsWrapper}>
        {renderButtons()}
      </div>
    </div>
  );
};


/**
 * PropTypes
 * @prop {Function} hideTableOfContents - Function to dispatch a
 * HIDE_TABLE_OF_CONTENTS action
 * @prop {Object} tableOfContentsState - The table of contents state in the
 * store
 */
TableOfContents.propTypes = {
  hideTableOfContents: PropTypes.func.isRequired,
  tableOfContentsState: PropTypes.object.isRequired,
};

export default TableOfContents;
