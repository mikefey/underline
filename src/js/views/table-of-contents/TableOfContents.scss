@import '~colors.scss';
@import '~variables.scss';

.wrapper {
  background-color: $white;
  display: none;
  float: left;
  height: 100%;
  position: fixed;
  width: 100%;
  z-index: 750;
}

.wrapperShowing {
  composes: wrapper;
  display: block;
}

.buttonsWrapper {
  float: left;
  height: 100%;
  max-width: $max-content-width;
  -webkit-overflow-scrolling: touch;
  overflow-y: scroll;
  padding: 60px 0 0;
  position: relative;
  width: 100%;
}
