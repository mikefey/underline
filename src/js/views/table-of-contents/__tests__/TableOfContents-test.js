import test from 'ava';
import React from 'react';
import { renderToString } from 'react-dom/server';

import TableOfContents from './../TableOfContents.jsx';

test('TableOfContents component', (t) => {
  const state = {
    data: [
      { label: 'item 1', href: 'item-1.html' },
      { label: 'item 2', href: 'item-2.html' },
    ],
    show: false,
  };
  const hideTableOfContents = () => {};

  const tree = renderToString(
    <TableOfContents
      hideTableOfContents={hideTableOfContents}
      tableOfContentsState={state}
    />,
  );

  t.snapshot(tree);
});
