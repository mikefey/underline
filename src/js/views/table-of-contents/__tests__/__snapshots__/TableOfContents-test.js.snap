// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableOfContents component 1`] = `"<div class=\\"TableOfContents__wrapper___2FkfbEMJUG4awYB\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"853761303\\"><div class=\\"Header__wrapper___2PTLf_nkW7S2HEU\\" data-reactid=\\"2\\"><div class=\\"Header__content___2VIdu565Foup5Pb\\" data-reactid=\\"3\\"><h2 class=\\"Header__title___1_ebCiqkttbS44m\\" data-reactid=\\"4\\">Table Of Contents</h2><button class=\\"CloseButton__wrapper___2i9TByxUopDp0cZ\\" data-reactid=\\"5\\"><div class=\\"CloseButton__icon___3Schb-DMASl5XVC\\" data-reactid=\\"6\\"><div class=\\"CloseButton__iconLineOne___1_bEHFLo8IfP5Fg\\" data-reactid=\\"7\\"></div><div class=\\"CloseButton__iconLineTwo___3dpb1YfW7QXAzBv CloseButton__iconLineOne___1_bEHFLo8IfP5Fg\\" data-reactid=\\"8\\"></div></div></button></div></div><div class=\\"TableOfContents__buttonsWrapper___3n-b4RZYhh_S3yB\\" data-reactid=\\"9\\"><button class=\\"TableOfContentsButton__wrapper___3vgI4LruZrYQ9H2\\" data-reactid=\\"10\\"><h4 data-reactid=\\"11\\">item 1</h4></button><button class=\\"TableOfContentsButton__wrapper___3vgI4LruZrYQ9H2\\" data-reactid=\\"12\\"><h4 data-reactid=\\"13\\">item 2</h4></button></div></div>"`;
